### PR TITLE
Add `Metadata` optional store trait,

### PR DESCRIPTION
**Pull request by panarch**
`Metadata` provides two methods which are currently matching `SHOW {VARIABLE};` queries.
- `schema_names`: `SHOW TABLES;`
- `version`: `SHOW VERSION;`

Add `Metadata` store traits and implement it for both `MemoryStorage` and `SledStorage`.

### CODE REVIEW

File: src/executor/execute.rs [hunk 1]
```diff
@@ -57,6 +60,16 @@ pub enum Payload {
     Commit,
     #[cfg(feature = "transaction")]
     Rollback,
+
+    #[cfg(feature = "metadata")]
+    ShowVariable(PayloadVariable),
+}
```
Comment (ever0de) [inline src/executor/execute.rs:66]: It has nothing to do with metadata, but how about attaching indexes to each other?

Comment (panarch) [reply src/executor/execute.rs:66]: I think... I don't get the point. Could you provide more details?

Comment (ever0de) [reply src/executor/execute.rs:66]: This is the sorting story inside `ExecuteError`.
It's related to formatting, so I don't think you need to think too much about it.

Group them in the form below.
```rust
     #[cfg(feature = "index")]
     CreateIndex,
     #[cfg(feature = "index")]
     DropIndex,

     #[cfg(feature = "transaction")]
     StartTransaction,
     #[cfg(feature = "transaction")]
     Commit,
     #[cfg(feature = "transaction")]
     Rollback,

     #[cfg(feature = "metadata")]
     ShowVariable(PayloadVariable),
}
```

File: tests/sled_transaction.rs [hunk 2]
```diff
@@ -729,3 +732,54 @@ async fn sled_transaction_timeout_index() {
         Ok(select!(id I64; 1))
     );
 }
+
+#[cfg(feature = "metadata")]
+#[test]
+fn sled_transaction_metadata() {
+    macro_rules! test_tables {
+        ($glue: ident []) => {
```
Comment (ever0de) [inline tests/sled_transaction.rs:740]: Q) [] What is syntax?
I wanted to study, but I haven't found the document yet.
Can you leave a reference?

**Review by ever0de — APPROVED** [review]
LGTM 👍 💯 🥇 🚀 👍 💯 🥇 🚀 👍 💯 🥇 🚀

**Pull request merged**